File: BJ/Solved/2579.swift
////
////  2579.swift
////  BJ
////
////  Created by 송재훈 on 9/28/24.
////
//
//
//let N = Int(readLine()!)!
//var stairs = [0]
//for _ in 1...N {
//    stairs.append(Int(readLine()!)!)
//}
//
//if N <= 2 {
//    print(N == 1 ? stairs[1] : stairs[1] + stairs[2])
//}
//else {
//    var dp = Array(repeating: Array(repeating: 0, count: 305), count: 305)
//    
//    (dp[1][1], dp[1][2]) = (stairs[1], 0)
//    (dp[2][1], dp[2][2]) = (stairs[2], stairs[1] + stairs[2])
//    
//    for i in 3...N {
//        dp[i][1] = max(dp[i - 2][1], dp[i - 2][2]) + stairs[i]
//        dp[i][2] = dp[i - 1][1] + stairs[i]
//    }
//    
//    print(max(dp[N][2], dp[N][1]))
//}

File: Python/Solved/2579.py
# 6
# 10
# 20
# 15
# 25
# 10
# 20

import sys
input = sys.stdin.readline

N = int(input())
stairs = [0] + [int(input()) for _ in range(N)]

if N <= 2: print(sum(stairs))
else:
    dp = [[0, 0] for _ in range(N + 1)]
    dp[1] = [stairs[1], 0]
    dp[2] = [stairs[2], stairs[1] + stairs[2]]

    for i in range(3, N + 1):
        dp[i][0] = max(dp[i - 2][0], dp[i - 2][1]) + stairs[i]
        dp[i][1] = max(dp[i - 1][0], dp[i - 2][1]) + stairs[i]
    print(max(dp[N]))